app/src/main/java/com/gh4a/utils/rx/RxTools.java

```diff
@@ -0,0 +1,102 @@
+package com.gh4a.utils.rx;
+
+import io.reactivex.Observable;
+import io.reactivex.ObservableTransformer;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+import android.app.Activity;
+import android.app.Dialog;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.design.widget.Snackbar;
+import android.support.v4.app.DialogFragment;
+import android.support.v4.app.FragmentActivity;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import com.gh4a.BasePagerActivity;
+import com.gh4a.R;
+import com.gh4a.utils.UiUtils;
+import com.philosophicalhacker.lib.RxLoader;
+
+public class RxTools {
+    public static <T> ObservableTransformer<T, T> handle(Activity activity, int id) {
+        return observable -> {
+            final RxLoader rxLoader = new RxLoader(activity, ((AppCompatActivity)activity).getSupportLoaderManager());
+            return observable
+                    .compose(handleError(activity))
+                    .compose(rxLoader.makeObservableTransformer(id));
+        };
+    }
+
+    public static <T> ObservableTransformer<T, T> handle(Activity activity, int id, boolean refresh) {
+        return observable -> {
+            final RxLoader rxLoader = new RxLoader(activity, ((AppCompatActivity)activity).getSupportLoaderManager());
+
+            if(refresh) {
+                return observable
+                        .compose(handleError(activity))
+                        .compose(rxLoader.makeObservableTransformer(id, refresh));
+            } else return observable.compose(handle(activity, id));
+        };
+    }
+
+    // Error handler showing SnackBar
+    public static <T> ObservableTransformer<T, T> onErrorSnackbar(Activity activity, int id, View rootLayout, String errorMessage, int messageRes) {
+        return observable -> {
+            final RxLoader rxLoader = new RxLoader(activity, ((AppCompatActivity)activity).getSupportLoaderManager());
+            return observable
+                    .compose(rxLoader.makeObservableTransformer(id))
+                    .doOnSubscribe(disposable -> showProgressDialog(activity, messageRes))
+                    .doOnTerminate(() -> dismissProgressDialog())
+                    .doOnError(error -> {
+                        Snackbar.make(rootLayout, errorMessage, Snackbar.LENGTH_LONG)
+                                .setAction(R.string.retry, (View.OnClickListener) activity)
```

Comment:
This still (I've commented on this before) is wrong: the click listener previously was ProgressDialogTask, and it re-scheduled the task. You now need to re-subscribe to the observable in that case.